### PR TITLE
Parallize late data copy

### PR DESCRIPTION
**Pull request by ydai1124**
@zliu41 Can you review?

### CODE REVIEW

File: gobblin-compaction/src/main/java/gobblin/compaction/mapreduce/MRCompactorJobRunner.java [hunk 1]
```diff
@@ -90,6 +97,9 @@
   private static final boolean DEFAULT_COMPACTION_JOB_OVERWRITE_OUTPUT_DIR = false;
   private static final String COMPACTION_JOB_ABORT_UPON_NEW_DATA = COMPACTION_JOB_PREFIX + "abort.upon.new.data";
   private static final boolean DEFAULT_COMPACTION_JOB_ABORT_UPON_NEW_DATA = false;
+  private static final String COMPACTION_COPY_LATE_DATA_THREAD_POOL_SIZE = COMPACTION_JOB_PREFIX
+      + "copy.latedata.thread.pool.size";
+  private static final int DEFAULT_COMPACTION_COPY_LATE_DATA_THREAD_POOL_SIZE = 1000;
```
Comment (zliu41) [inline gobblin-compaction/src/main/java/gobblin/compaction/mapreduce/MRCompactorJobRunner.java]: Is 1000 threads not too much for a single JVM?

Comment (ydai1124) [inline gobblin-compaction/src/main/java/gobblin/compaction/mapreduce/MRCompactorJobRunner.java]: That's fine for Az machine. But reduced the default to 20 now.

File: gobblin-compaction/src/main/java/gobblin/compaction/mapreduce/MRCompactorJobRunner.java [hunk 2]
```diff
@@ -270,14 +285,34 @@ private DateTime getCompactionTimestamp() throws IOException {
...
-  private void copyDataFiles(Path outputDirectory, List<Path> inputFilePaths) throws IOException {
-    for (Path filePath : inputFilePaths) {
-      Path convertedFilePath = this.outputRecordCountProvider
-          .convertPath(this.lateInputRecordCountProvider.restoreFilePath(filePath), this.inputRecordCountProvider);
-      String targetFileName = convertedFilePath.getName();
-      Path outPath = this.lateOutputRecordCountProvider.constructLateFilePath(targetFileName, this.fs, outputDirectory);
-      HadoopUtils.copyPath(this.fs, filePath, this.fs, outPath, this.fs.getConf());
-      LOG.info(String.format("Copied %s to %s.", filePath, outPath));
+  private void copyDataFiles(final Path outputDirectory, List<Path> inputFilePaths) throws IOException {
+    ListeningExecutorService executor =
```
Comment (zliu41) [inline gobblin-compaction/src/main/java/gobblin/compaction/mapreduce/MRCompactorJobRunner.java]: executor needs to be closed

Comment (zliu41) [inline gobblin-compaction/src/main/java/gobblin/compaction/mapreduce/MRCompactorJobRunner.java]: Also no need to use ListeningExecutor

**Comment by ydai1124** [comment]
@zliu41  updated.

**Comment by zliu41** [comment]
LGTM

**Pull request merged**